### ISSUE TITLE
bug in function to get application_id ?

### ISSUE THREAD

**Issue by cvdabbeele** [cloudconformity/azure-onboarding-scripts#21]
Hi,
IMHO, the function to get the application_id is missing `--all`  

```bash
get_application_id_of_default_app_registration() {
  application_id=$(az ad app list --all --query "[?displayName == '${DEFAULT_APP_REGISTRATION_NAME}'].appId" --output tsv)
}
```
Without this option the returned list is incomplete and in my case it had not seen an existing app registration
Also, at runtime the script mentions the following error:
```bash
bash apply-roles 
This script will create the necessary resources for Cloud One Conformity to scan your Azure subscriptions.
Would you like to use the default App Registration "Conformity Azure access" [Y|N]? y
Checking if an App Registration with the name "Conformity Azure access" exists...
WARNING: The result is not complete. You can still use '--all' to get all of them with long latency expected, or provide a filter through command arguments
App Registration "Conformity Azure access" will be created as it wasn't found.
Would you like to auto generate the Client secret key for your new App Regi
```
thank you,
chrisV

**Comment by iaindurham** [comment]
Hi @cvdabbeele thanks for raising this issue. 

This will be resolved in https://github.com/cloudconformity/azure-onboarding-scripts/pull/25

As an FYI this is actually caused by this behaviour in the Azure CLI https://github.com/Azure/azure-cli/issues/9486

**Comment by iaindurham** [comment]
@cvdabbeele This issue has been resolved and is now in the `master` branch and is included in the `v3.0.0` release.
